ArduCopter/AP_Arming.cpp

```diff
@@ -531,12 +533,11 @@ bool AP_Arming_Copter::mandatory_gps_checks(bool display_failure)
         if (reason == nullptr || strlen(reason) == 0) {
             if (!mode_requires_gps && fence_requires_gps) {
                 // clarify to user why they need GPS in non-GPS flight mode
-                reason = "Fence enabled, need 3D Fix";
+                check_failed(display_failure, "Fence enabled, need GPS Fix %u", gps.status_arm_min());
             } else {
-                reason = "Need 3D Fix";
+                check_failed(display_failure, "Needs GPS Fix %u", gps.status_arm_min());
             }
         }
-        check_failed(display_failure, "%s", reason);
```

Comment:
Could you leave this in so that if reason is not null or empty, it will get called? Wrap it in an else to avoid calling it twice if reason was previously null?

Reply:
Done